src/charm.py

```diff
@@ -711,6 +729,23 @@ def _on_start(self, event: StartEvent) -> None:
         # Bootstrap the cluster in the leader unit.
         self._start_primary(event)
 
+    def _setup_exporter(self) -> None:
+        """Set up postgresql_exporter options."""
+        cache = snap.SnapCache()
+        postgres_snap = cache[POSTGRESQL_SNAP_NAME]
+
+        try:
+            postgres_snap.set(
+                {
+                    "exporter.user": MONITORING_USER,
+                    "exporter.password": self.get_secret("app", MONITORING_PASSWORD_KEY),
+                }
+            )
+            postgres_snap.restart(services=[MONITORING_SNAP_SERVICE])
+
+        except snap.SnapError:
+            logger.exception("failed to set up postgresql_exporter options")
```

Comment:
That's a good question and I tend to agree that go to blocked is a better approach, since this is failing silently and will just postpone user frustration.

Reply:
3227f95